enterprise_subsidy/apps/subsidy/models.py

```diff
@@ -416,6 +419,7 @@ def commit_transaction(self, ledger_transaction, fulfillment_identifier=None, ex
             ledger_transaction.external_reference.set([external_reference])
         ledger_transaction.state = TransactionStateChoices.COMMITTED
         ledger_transaction.save()
+        event_bus.send_transaction_committed_event(ledger_transaction)
```

Comment:
If the event bus send fails for these calls is the `ledger_transaction` also rolled back? 

I think my overall question is this: When these event emissions fail to send, should the related DB writes also fail?

Reply:
No, because then we're making the communication pattern synchronous instead of asynchronous.

Reply:
But if the event doesn't make it on to the bus, and the record is written systems downstream of this message will not be able to act accordingly. Not having this event hit the bus means that whatever actions were supposed to happen because of this do not happen leaving the system in an inconsistent state. 

I don't believe this makes it synchronous. It's more that the transaction isn't complete until the event is sent.

Reply:
I would also suggest that this does not make it synchronous either as we are simply writing that the event happened. Not waiting for downstream systems to execute on the message. 

I think that writing these messages on to the bus and ensuring that call succeeds is just as important now as ensuring the DB record is written.

Reply:
Synchronous was the wrong word, it's more like you'd be making the send of the message part of the "atomic block" of the flow.
Part of the goal of using an async event broker, though, is to move toward "eventual consistency", and force our serves to adapt to inconsistent states. The ideal thing to do if kafka is down/unreachable/whatever isn't to fail the redemption flow; it's to have a mechanism for replaying events that occurred in a given time window, and to be able to reply on idempotent consumers to process those replayed events. Or to implement a dead-letter queue (here on the producer side).

Reply:
Now, having said all that, I should go through and check if this is already called from within a `transaction.atomic()` block.